tempo_embeddings/embeddings/vector_database.py

```diff
@@ -0,0 +1,336 @@
+            "requires_explicit_embeddings": self.requires_explicit_embeddings,
+            "existing_collections": []
+        }
+    
+    def _save_config(self):
+        if self.config:
+            json.dump(self.config, open(f"{self.db_path}/config.json", "w"), indent=2)
+
+    def connect(self):
+        if not self.client:
+            self.client = chromadb.PersistentClient(path=self.db_path)
+            # If the given path existed then we will load the records AND override the parameters with the config
+            config_path = f"{self.db_path}/config.json"
+            if os.path.exists(config_path):
+                self.config = json.load(open(config_path))
+                print(f"Database Path '{self.db_path}' already exists. Loading current configuration\n{self.config}")
+            else:
+                print(f"Creating NEW Database in {self.db_path}...")
+                self._save_config()
+        else:
+            print(f"A connection to the client already exsists in this session: {self.client}")
+
+    def create_new_collection(self, name: str, passages: List[Passage] = None, embeddings: ArrayLike = None) -> Optional[Collection]:
+        if not self.client:
+            print("Please connect to a valid database first")
+            return None
+        # Create NEW collection and Embeds the given passages. Do nothing otherwise
+        try:
+            if self.embedding_function:
+                collection = self.client.create_collection(name, 
+                                                            embedding_function=self.embedding_function,
+                                                            metadata={"hnsw:space": "cosine"})
+            else:
+                collection = self.client.create_collection(name, metadata={"hnsw:space": "cosine"})
+            self.config["existing_collections"].append(name)
+            self.active_collection_name = name
+            self._save_config()
+            # If the collection is new then insert the corresponding passages already
+            if passages: self.insert_passages_embeddings(collection, passages, embeddings)
+
+            return collection
+        except UniqueConstraintError:
+            print(f"Collection '{name}' already exists. Not doing anything. (You can delete it or use the 'get_existing_collection' method)")
+            return None
+        
+    def get_existing_collection(self, name: str) -> Optional[Collection]:
+        try:
+            collection = self.client.get_collection(name, embedding_function=self.embedding_function)
+            self.active_collection_name = name
+            return collection
+        except:
+           return None
+
+
+    def delete_collection(self, name):
+        try:
+            self.client.delete_collection(name)
+            print(f"Succesfully deleted {name}")
+        except:
+            print(f"Collection '{name}' does not exist in this database.")
```

Comment:
Catching all exceptions is quite risky because this implicitly handles exceptions that you might not have expected here at all. In this case, I would tend to not catching any exception, hence pass them on to the user (unless we know specifically which exceptions to expect and how to handle them).